### ISSUE TITLE
Thumbnail Viewer for assets during download and cache

### ISSUE THREAD

**Issue by dsync89** [syd711/vpin-studio#618]
First of all, great job in this convenient vpin table management tool that ease the table management and media asset audit all-in-one! Just some QOL suggestions that might improve the overall usage even more.

I've previously working with a [pinup popper frontend media auditor](https://github.com/dsync89/fuzzymatch-pinball-media) that focuses on auditing media, in particular, allowing selection of asset such as wheels and present them in thumbnail fashion, not sure if we could have similar featuer here in vpin studio? Most importantly, cache the result for that asset type if the image had been loading before during that selection. Sometimes I find myself selecting back and forth between two wheels (Raja) and (Celium) wheels, and the loading time is noticeable even though the PNG is only KBs in size, not sure what causes the loading delay /hmm

Some screenshot from my Media Auditor:
![image](https://github.com/user-attachments/assets/1666fc0d-b3d4-4cd3-9971-1dfa147d80fc)

**Comment by syd711** [comment]
Hi @dsync89, thanks for you feedback!
Not sure if I got the use case right. Yes, we can provide a thumbnail view for all assets of a screen. But the goal to to rename or find unused ones? What exactly should happen after the selection?

**Comment by dsync89** [comment]
Hi @syd711 thanks for replying to the feat request.

To clarify, the thumbnail feature would benefit the workflow of downloading/selecting remote assets to download. The goal is to allow user to select which remote asset to download for a given asset type (e.g. wheels, playfield video etc.), especially given that some asset type might have several images. Currently my workflow involve clicking those texts one by one, wait a while for the image preview to show, then move on to the next. Sometimes the selection might ended up being a tie, and the non-caching mechanism so far makes A/B selection harder since I had to go back and forth between those two selections, perhaps I'm just OCD :)

**Comment by syd711** [comment]
Yes, the request totally makes sense and would improve the usability. The issue here is that this would mean to pull all assets for every screen selection. We rely on the infrastructure from PinUP Popper here and are grateful that we got access to it. But I know that nailbuster is really careful when it comes to additional traffic on his servers. I get that, because he offers this service for free.
So unfortunately I have to close this ticket as a won't fix, since this would significantly increase the traffic/costs for him.
